hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -178,11 +178,6 @@ private[hudi] object HoodieSparkSqlWriter {
             } else {
               hoodieAllIncomingRecords
             }
-
-          if (hoodieRecords.isEmpty()) {
```

Comment:
Hi @pengzhiwei2018  I think this is necessary. It is possible that the incoming rdd is empty and we need to handle that. Are you seeing any performance issues here?
Worth to mention that this is the first Spark action, if you are seeing isEmpty() take too long in Spark UI, it's probably the transformation before this action.

Reply:
Hi @garyli1019 ,In our case,the incoming rdd is very complex, this `isEmpty` test spend so much time.And if the incoming rdd is empty, the code that follows will execute quickly.So I think this `isEmpty` test is unnecessary.

Reply:
Hi @pengzhiwei2018 , I guess this `isEmpty()` triggerred the complex rdd transformations, the actual time consuming part is the transformation before this check. We can double check by put something like `show(1)` before the `isEmpty` check, then the time-consuming part should become the `show(1)` and the `isEmpty` check should finish fast. 
IIRC, an empty rdd will trigger an error later if we don't have this check here. If that's not the case, would you write a unit test to verify?

Reply:
Hi @garyli1019 ,Thanks for you suggestion.I will add some test case later.